### PR TITLE
Use 3x2 matrix for coordinate transformation

### PR DESCRIPTION
**Pull request by ekrctb**
For coordinate transformation matrices `DrawInfo.Matrix` and `InverseMatrix`, its third column is always `[0 0 1]^T`. Thus we can omit this column. I think it is good because it is more no-implicit-assumption and potentially less space/computation.

---

### CODE REVIEW

**Comment by Tom94** [comment]
I intentionally left the last column in to support perspective transforms for potential 2.5D effects in the future.

**Comment by peppy** [comment]
Yep, will definitely profile before merging this. I foresee it helping substantially, though.

File: osu.Framework/Graphics/Drawable.cs [hunk 1]
```diff
@@ -1490,7 +1490,7 @@ private DrawInfo computeDrawInfo()
 
                 // Cannot use ToParentSpace here, because ToParentSpace depends on DrawInfo to be completed
                 // ReSharper disable once PossibleNullReferenceException
-                Quad interp = Quad.FromRectangle(DrawRectangle) * (di.Matrix * Parent.DrawInfo.MatrixInverse);
+                Quad interp = Quad.FromRectangle(DrawRectangle) * di.Matrix * Parent.DrawInfo.MatrixInverse;
```
[the remark this answers is not in the document]
Comment (ekrctb) [reply osu.Framework/Graphics/Drawable.cs]: A transform (1x2 vec * 3x2 mat) requires 4 multiplications and for a quad 16 multiplications.
3x3 matmul, 18 multiplications are required.
3x2 pseudo matmul (3x3 matmul assuming third col is [0 0 1]) requires 12 multiplications.

two quad transforms: 16 * 2 = 32.
two vector transforms: 4 * 2 = 8.

When matrix is 3x3:
  first matmul then transform quad: 18 + 16 = 34.
  first matmul then transform a vec: 18 + 4 = 22.

When matrix is 3x2:
  first matmul then transform quad: 12 + 16 = 28.
  first matmul then transform a vec: 12 + 4 = 16.

So when only matrix is 3x2 and for quad transform, first matmul is better.
But I thought it is good to doing the same way for quad transform as vec transform due to floating point error.

[the remark this answers is not in the document]
Comment (Tom94) [reply osu.Framework/Graphics/Drawable.cs]: For floating point numbers additions are actually pretty much as bad as multiplications, so you should not disregard them. That said, I don't think you need to worry about floating point error here. It's pretty much negligible as long as you're not running into something like catastrophic cancellation (which would require division to happen).

**Comment by peppy** [comment]
Testing on osu!, there doesn't look to be any discernible performance gains. @ekrctb did you happen to benchmark `Matrix3` vs `Matrix3x2` at your end?

**Comment by ekrctb** [comment]
Closing this because I think it is more productive to do this when necessary / where necessary.

A note for perspective projection transformation: some code assuming a local transformation is linear. For example, for a non-linear transformation, `Position` difference causes a bounding box size difference. Thus we don't want to add thing like `Perspective` at `Drawable` level.

**Pull request closed without merging**